### PR TITLE
SKIP_IF and XFAIL_IF errors should not cause the run to abort

### LINKED ISSUE #205: Bad skip condition definition can abort entire test run
The [doctest code that decides whether a test is skipped](https://github.com/catch22/octave-doctest/blob/55a8f089b2f581f5134532fad9431eefeb4a5b68/inst/private/doctest_run_tests.m#L47-L51) is an unprotected eval.

```
  % determine whether test should be skipped
  % (careful about Octave bug #46397 to not change the current value of “ans”)
  eval (strcat ('DOCTEST__current_test.skip = ', ...
                 doctest_join_conditions(DOCTEST__current_test.skip), ...
                ';'));
```

This means that an error in a tested file's skip condition definition will bubble up to the top, and abort the entire test run. This condition occurs with the doctest code in the Octave Forge interval and symbolic packages.

Maybe this `eval()`, and the others in `doctest_run_tests`, should be protected by a try/catch that turns a bad skip or xfail definition into a test failure, instead of aborting the run?

Caused https://github.com/apjanke/octave-testify/issues/44.

### PR DESCRIPTION
**Pull request by cbm755**
Fixes #205.  Add some tests but they will cause "make test" to fail.
For now just put them in a `.m.txt`, which will stop doctest from
finding them.  Hopefully can address this as part of the BIST effort
elsewhere.

### CODE REVIEW

**Comment by cbm755** [comment]
I'm not sure about this: its kind of a hack, because I'm putting the error message into `got` (which is the text that is compared to `want`). 

The proper thing to do is probably to handle this up one or two levels in `doctest_collect`.  But maybe this is good enough?

**Comment by cbm755** [comment]
I think #210 would give me a way to make the test work properly.

File: inst/private/doctest_run_tests.m [hunk 1]
```diff
@@ -40,25 +40,53 @@
...
-  eval (strcat ('DOCTEST__current_test.skip = ', ...
-                 doctest_join_conditions(DOCTEST__current_test.skip), ...
-                ';'));
+  try
+    eval (strcat ('DOCTEST__current_test.skip = ', ...
+                  doctest_join_conditions(DOCTEST__current_test.skip), ...
+                  ';'));
+  catch DOCTEST__exception
+    DOCTEST__current_test.directive_issue = true;
+    DOCTEST__current_test.skip = [];
```
Comment (catch22) [inline inst/private/doctest_run_tests.m]: `[]` should probably be `false`

File: inst/private/doctest_run_tests.m [hunk 2]
```diff
@@ -40,25 +40,53 @@
   doctest_datastore('set_current_index', DOCTEST__i);
   DOCTEST__current_test = doctest_datastore('get_current_test');
 
+  DOCTEST__current_test.directive_issue = false;
```
Comment (catch22) [inline inst/private/doctest_run_tests.m]: As far as I can see, we are not actually ever using this variable. Perhaps let's remove this for now until we replace this by a non-hacky implementation?

File: inst/private/doctest_run_tests.m (hunk 1 above)
Comment (cbm755) [reply inst/private/doctest_run_tests.m]: Or maybe `true`: the test is not run: as far as future tests are concerned it was skipped...

**Comment by cbm755** [comment]
I will merge for now when tests pass; can always be refactored later.

**Pull request merged**